src/network/serverpackethandler.cpp

```diff
@@ -629,13 +629,21 @@ void Server::handleCommand_InventoryAction(NetworkPacket* pkt)
 		if (ma->from_inv != ma->to_inv)
 			m_inventory_mgr->setInventoryModified(ma->to_inv);
 
-		bool from_inv_is_current_player =
-			(ma->from_inv.type == InventoryLocation::PLAYER) &&
-			(ma->from_inv.name == player->getName());
-
-		bool to_inv_is_current_player =
-			(ma->to_inv.type == InventoryLocation::PLAYER) &&
-			(ma->to_inv.name == player->getName());
+		bool from_inv_is_current_player = false;
+		if (ma->from_inv.type == InventoryLocation::PLAYER) {
+			if (ma->from_inv.name != player->getName())
+				return;
+			from_inv_is_current_player = true;
+		} else if (ma->from_inv.type != InventoryLocation::CURRENT_PLAYER)
```

Comment:
should be `ma->to_inv.type != InventoryLocation::CURRENT_PLAYER`
you can also move that to the same line as the declaration (`bool foo = x == y;`)